docs/0.5/howto/configure_logging.md

```diff
@@ -0,0 +1,230 @@
+
+[loggers.root]
+appenders = ["stdout"]
+level = "Trace"
+
+[loggers.tokio]
+appenders = ["stdout"]
+level = "Warn"
+
+[loggers.tokio_reactor]
+appenders = ["stdout"]
+level = "Warn"
+
+[loggers.hyper]
+appenders = ["stdout"]
+level = "Warn"
+
+```
+
+### Redirect Splinter Daemon logs to file
+
+This example will redirect the logs from libsplinter and the splinter daemon to
+a single log file. This is the most generally useful configuration besides the
+default configuration because it will seperate the logs related to splinters
+functionality from the logs generated by underlying libraries.
+
+``` console
+
+[appenders.splinter_file]
+kind = "file"
+filename = "/var/log/splinter/splinter.log"
+
+[loggers.splinter]
+appenders = ["splinter_file"]
+level = "Info"
+
+[loggers.splinterd]
+appenders = ["splinter_file"]
+level = "Info"
+
+```
+
+### All Logs All the Time
+
+This example has turned off all log filtering, every log statement will be
+logged to stdout. This is not recommended and may cause performance issues.
+
+``` console
+
+[loggers.root]
+appenders = ["stdout"]
+level = "Trace"
+
+[loggers.tokio]
+level = "Trace"
+
+[loggers.tokio_reactor]
+level = "Trace"
+
+[loggers.hyper]
+level = "Trace"
+
+```
+
+
+## Procedure
+
+
+1. Set up Appenders.
+
+    a. Set appender name `[appenders.<appender_name_here>]`.
+
+    b. Select appender kind to match your use case. `kind = "<appender_kind>"`
+
+    c. Fill other required fields dependant on the appender kind. Filename is
+    required for file and rolling_file kinds while size is only required for
+    rolling_file appenders.
+
+    d. Repeat steps a-c for any other appenders needed.
+
+
+2. Set up Loggers.
+
+   a. Set logger name `[loggers.<logger_name_here>]`, remembering that the name
+   determines which Rust modules will write to this logger.
+
+   b. Set appenders field to list of appender targets. `appenders =
+   ["<appender_names_configured above>"]`
+
+   c. Optionally set the Level field to filter the log messages forwarded to the
+   appenders. `level = "Info"`
+
+   d. Repeat steps a-c for each module that needs logging configuration.
+
+3. (Optional) Override default loggers. Any logger named `root` will override
+   the default included default logger.
+
+``` console
+[loggers.root]
+appenders = ["stdout", "file"]
+level = "Info"
+```
+
+The above example will set the root logger to log all messages with level
+`Info` and above to the `stdout`, and `file` appender.
+
+The `tokio`, `tokio_reactor`, and `hyper` loggers are included by default and
+will log to the `stdout` appender. All of them can be overridden as shown in
+the Example Configurations section above.
+
+## Troubleshooting
+
+Appender names have to exactly match between the logger appenders field and the
+appender definition.
+
+Setting up loggers to capture logs from sub modules requires that the names be
+quoted. For example `[loggers."splinterd::config"]` to select for the
+splinterd::config module.
+
```

Comment:
Remove the empty line at the end of the file

Reply:
fixed